clients/spark/core/src/main/scala/io/treeverse/clients/Exporter.scala

```diff
@@ -38,38 +45,51 @@ class Exporter(spark : SparkSession, apiClient: ApiClient, repoName: String, dst
     val newDF = LakeFSContext.newDF(spark, repoName, commitID)
     val prevDF = LakeFSContext.newDF(spark, repoName, prevCommitID)
 
-    newDF.createOrReplaceTempView("new_commit")
-    prevDF.createOrReplaceTempView("prev_commit")
+    val gen = Random.alphanumeric.dropWhile(_.isDigit)
+    val newTableName =  gen.take(8).mkString("") +"_new_commit"
+    val prevTableName =  gen.take(8).mkString("") +"_prev_commit"
+    newDF.createOrReplaceTempView(newTableName)
+    prevDF.createOrReplaceTempView(prevTableName)
 
     // pin Exporter field to avoid serialization
     val dst = dstRoot
 
-    val actionsDF = spark.sql("WITH raw_rows AS (SELECT n.key as nkey, n.address as naddress, n.etag as netag, " +
-      "p.key as pkey, p.address as paddress, p.etag as petag " +
-      "FROM new_commit n " +
-      "FULL OUTER JOIN prev_commit p " +
-      "ON n.key = p.key " +
-      "WHERE n.etag <> p.etag OR n.etag is null or p.etag is null) " +
-      "SELECT " +
-      "CASE WHEN nkey is null THEN 'delete' ELSE 'copy' END as action, " +
-      "CASE WHEN nkey is null THEN pkey ELSE nkey END as key, " +
-      "CASE WHEN naddress is null THEN paddress ELSE naddress END as address, " +
-      "CASE WHEN netag is null THEN petag ELSE netag END as etag ")
-
-    export(ns, dst, actionsDF)
-    spark.sparkContext.stop()
+    val actionsDF = spark.sql(s"""
+    SELECT
+      CASE WHEN nkey is null THEN 'delete' ELSE 'copy' END as action,
+      CASE WHEN nkey is null THEN pkey ELSE nkey END as key,
+      CASE WHEN naddress is null THEN paddress ELSE naddress END as address,
+      CASE WHEN netag is null THEN petag ELSE netag END as etag
+    FROM
+    (SELECT n.key as nkey, n.address as naddress, n.etag as netag,
+      p.key as pkey, p.address as paddress, p.etag as petag
+      FROM ${newTableName} n
+      FULL OUTER JOIN ${prevTableName} p
+      ON n.key = p.key
+      WHERE n.etag <> p.etag OR n.etag is null or p.etag is null)
+    """)
+
+    export(ns, dst, actionsDF, false)
+    export(ns, dst, actionsDF, true)
   }
 }
 
 object Exporter {
-  private def handleRow(ns: String, rel: String, serializedConf: SerializableWritable[org.apache.hadoop.conf.Configuration], row: Row): Unit =  {
+  val sparkSuccessFileSuffix = "_SUCCESS"
```

Comment:
This probably needs to be a configuration parameter.  `_SUCCESS` is a great default, but I'm sure other values will be useful.  Also I expect a regexp (or other rule engine) to be a better choice -- e.g. I can export success files last _in some directories_

Reply:
Deferred for now, makes the code cumbersome and I need to think of a way to pass this in an orderly manner.